docs/reference/openapi-rules.md

```diff
@@ -1,3 +1,546 @@
+  /pet:
+    patch:
+      operationId: "update-pet"
+      responses: 
+        200:
+          description: ok
+    put:
+      operationId: "replace-pet"
+      responses: 
+        200:
+          description: ok
+```
+
+### operation-parameters
+
+Operation parameters are unique and non-repeating.
+
+1. Operations must have unique `name` + `in` parameters.
+2. Operation cannot have both `in: body` and `in: formData` parameters. (OpenAPI v2.0)
+3. Operation must have only one `in: body` parameter. (OpenAPI v2.0)
+
+**Recommended:** Yes
+
+### path-params
+
+Path parameters are correct and valid.
+
+1. for every param referenced in the path string (i.e: `/users/{userId}`), the parameter must be defined in either
+   `path.parameters`, or `operation.parameters` objects.
+
+2. every `path.parameters` and `operation.parameters` parameter must be used in the path string
+
+**Recommended:** Yes
+
+### contact-properties
+
+The [info-contact](#info-contact) rule will ask you to put in a contact object, and this rule will make sure it's full of the most useful properties: `name`, `url` and `email`. 
+
+Putting in the name of the developer/team/department/company responsible for the API, along with the support email and help-desk/GitHub Issues/whatever URL means people know where to go for help. This can mean more money in the bank, instead of developers just wandering off or complaining online.
+
+**Recommended:** No
+
+**Good Example**
+
+```yaml
+openapi: "3.0.2"
+info:
+  title: Awesome API
+  description: A very well defined API
+  version: "1.0"
+  contact:
+    name: A-Team
+    email: a-team@goarmy.com
+    url: goarmy.com/apis/support
+```
+
+### example-value-or-externalValue
+
+Examples for `requestBody` or response examples can have an `externalValue` or a `value`, but they cannot have both.
+
+**Recommended:** No
+
+**Bad Example**
+
+```yaml
+paths:
+  /pet:
+    put:
+      operationId: "relace-pet"
+      requestBody:
+        content:
+          'application/json':
+            examples: 
+              foo:
+                summary: A foo example
+                value: {"foo": "bar"}
+                externalValue: 'http://example.org/foo.json' 
+                # marp! no, can only have one or the other
+```
+
+### info-contact
+
+Info object should contain `contact` object. 
+
+Hopefully your API description document is so good that nobody ever needs to contact you with questions, but that is rarely the case. The contact object has a few different options for contact details.
+
+**Recommended:** Yes
+
+**Good Example**
+
+```yaml
+openapi: "3.0.2"
+info:
+  title: Awesome API
+  version: "1.0"
+  contact: 
+    name: A-Team
+    email: a-team@goarmy.com
+```
+
+### info-description
+
+OpenAPI object info `description` must be present and non-empty string.
+
+Examples can contain Markdown so you can really go to town with them, implementing getting started information like where to find authentication keys, and how to use them.
```

Comment:
"go to town" nice, added to my dictionary! :D is that British English? 

General question: what do you think about using slang phrases vs "general English"? Some other company I worked before (British company) had a guideline suggesting to avoid any form of "jargon" (whether that's technical, cultural or any other).

Reply:
There is no company guideline but we're just being a bit more human and fun in a document which is rather boring and terse otherwise.